src/js/index.ts

```diff
@@ -441,6 +442,54 @@ function inDepthBookmarkTree(toTreat: chrome.bookmarks.BookmarkTreeNode, parentP
   }
 }
 
+function toggleAgenda() {
+  if (OPTS.agendaUrl === chrome.i18n.getMessage('default_agenda_link')) {
+    toast.html('agenda', chrome.i18n.getMessage('no_agenda_link'));
+    return;
+  }
+  let panel = els.main.querySelector('#agendaPanel');
+  if (!panel) {
+    panel = createPanel(els.main);
+    panel.id = 'agendaPanel';
+    panel.firstElementChild!.textContent = chrome.i18n.getMessage('agenda');
+  }
+  updateAgenda();
+  if (panel.classList.contains('folded')) panel.classList.toggle('folded');
+  let e = panel.parentElement;
+  while (e && e !== els.main) {
+    if (e.classList.contains('folded')) e.classList.toggle('folded');
+    e = e.parentElement;
+  }
+  panel.scrollIntoView({ behavior: 'smooth' });
+  flash(panel as HTMLElement, 'highlight');
+}
+
+function updateAgenda() {
+  if (OPTS.agendaUrl === chrome.i18n.getMessage('default_agenda_link')) return;
+  const rootPanel = els.main.querySelector('#agendaPanel') as HTMLElement;
+  if (!rootPanel) return;
+  while (rootPanel.lastElementChild!.firstChild) {
+    rootPanel.lastElementChild!.removeChild(rootPanel.lastElementChild!.lastChild!);
+  }
+  const xmlHttp = new XMLHttpRequest();
```

Comment:
XMLHttpRequest is old.
Please use fetch, examples here: https://github.com/portsoc/fetch101